src/multimedia.c

```diff
@@ -276,25 +281,151 @@ int cropRGB24(unsigned char *inputRGB24, int width, int height, int startX, int
 
 }
 
-static void process_image(const void *p, int size, char* output_filestring, char* output_cropped_filestring, crop_window c_window)
+/*
+*  Function: RGB24toGrayscale
+*  --------------------------
+*
+*  inputRGB24       Pointer to the bytes in memory holding the pixels of the RGB24 bitmap.
+*  width            Width in pixels of the input RGB24 bitmap.
+*  height           Height in pixels of the input RGB24 bitmap.
+*  outputGrayscale  Pointer to the bytes in memory which will hold the pixels of the 8-bit
+*                   grayscale bitmap.
+*
+*  This function calculates the luminance Y of each pixel of the input RGB24 bitmap, and
+*  clamps its value to the 0 - 255 range. It then assigns this value to the corresponding
+*  pixel of the output grayscale bitmap.
+*/
+int RGB24toGrayscale(unsigned char *inputRGB24, int width, int height, unsigned char *outputGrayscale)
+{
+    unsigned int i, j;
+    unsigned int pitchInputRGB, pitchOutputGrayscale;
+    unsigned char* pMovInputRGB;
+    unsigned char* pMovOutputGrayscale;
+
+    pitchInputRGB = ALIGN_TO_FOUR(3*width);
+    pitchOutputGrayscale = ALIGN_TO_FOUR(width);
+    pMovOutputGrayscale = outputGrayscale;
+
+    for(j=0; j < height; j++) {
+        for(i=0; i < width; i++) {
+            int R, G, B, Y;
+            pMovInputRGB = inputRGB24 + j*pitchInputRGB + i*3;
+            pMovOutputGrayscale = outputGrayscale + j*pitchOutputGrayscale + i;
+
+            B = *pMovInputRGB;
+            G = *(pMovInputRGB + 1);
+            R = *(pMovInputRGB + 2);
+
+            Y = 0.2126 * (double)R  + 0.7152 * (double)G + 0.722 * (double)B;
```

Comment:
What are these fractional values?
```
> 0.2126
> 0.7152
> 0.722
```

Reply:
Luminance Y is calculated from RGB in a variety of different ways, leading to different answers. The way chosen depends on the application/image etc. I used the coefficients found here on [stackoverflow](https://stackoverflow.com/questions/687261/converting-rgb-to-grayscale-intensity), but I dropped the gamma correction because the images were completely whiting out. A survey of different ways of calculating the grayscale is carried out [here](http://cadik.posvete.cz/color_to_gray_evaluation/).

I just needed a way to visualize my grayscale and have it look kind of gray. So I just threw something together that worked. Yeah I could go deeper into how/why ... but seems like a distraction at this point. Until it is a problem and I need to ofcourse.